src/hooks/use-token-signer.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import {useRouter} from 'next/router'
+import axios from 'axios'
+import cookie from '../utils/cookies'
+import {AUTH_DOMAIN} from '../utils/auth'
+
+const http = axios.create()
+
+function getSingleQueryValue(
+  param: undefined | string | string[],
+): string | undefined {
+  if (Array.isArray(param)) {
+    return param[0]
+  } else {
+    return param
+  }
+}
+
+function useTokenSigner(): void {
```

Comment:
If the referral token is in the URL's query params, grab it, request that the server sign it, then stick the signed value in a cookie.

If a signed referral token is already present, don't do any of this.

Reply:
nitpick (typing): return `void` is redundant and always (safely) inferred